File: Typescript/src/finalPositionOfSnake.ts
export function finalPositionOfSnake(n: number, commands: string[]): number {
	let x = 0;
	let y = 0;
	for (const e of commands) {
		switch (e) {
			case "UP":
				x--;
				break;
			case "DOWN":
				x++;
				break;
			case "LEFT":
				y--;
				break;
			case "RIGHT":
				y++;
				break;
		}
	}
	return x * n + y;
};

File: Dart/lib/finalPositionOfSnake.dart
class Solution {
  int finalPositionOfSnake(int n, List<String> commands) {
    var pos = 0;
    for (var command in commands) {
      if (command == "RIGHT") {
        pos++;
      } else if (command == "LEFT") {
        pos--;
      } else if (command == "UP") {
        pos -= n;
      } else {
        pos += n;
      }
    }
    return pos;
  }
}
